cmd/aws-iam-authenticator/add.go

```diff
@@ -73,18 +85,30 @@ var addRoleCmd = &cobra.Command{
 	Short: "add a role entity to an existing aws-auth configmap, not for CRD/file backends",
 	Long:  "NOTE: this does not currently support the CRD and file backends",
 	Run: func(cmd *cobra.Command, args []string) {
-		if roleARN == "" || userName == "" || len(groups) == 0 {
+		if (roleARN == "" && roleARNLike == "") || userName == "" || len(groups) == 0 {
 			fmt.Printf("invalid empty value in rolearn %q, username %q, groups %q", roleARN, userName, groups)
 			os.Exit(1)
 		}
 
-		checkPrompt(fmt.Sprintf("add rolearn %s, username %s, groups %s", roleARN, userName, groups))
+		var arnOrArnLike string
+		switch {
+		case roleARN != "" && roleARNLike != "":
+			fmt.Printf("only one of --rolearn or --rolearnlike can be supplied")
+			os.Exit(1)
+		case roleARN != "":
+			arnOrArnLike = "userarn"
+		case roleARNLike != "":
+			arnOrArnLike = "userarnlike"
```

Comment:
these should be `rolearn` and `rolearnlike` respectively, right?

Reply:
Nice catch!